mobile/lib/services/preview_video_store.dart

```diff
@@ -0,0 +1,525 @@
+        final header = response.data["data"]["decryptionHeader"];
+        final encryptionKey = getFileKey(file);
+        final playlistData = await decryptAndUnzipJson(
+          encryptionKey,
+          encryptedData: encryptedData,
+          header: header,
+        );
+        finalPlaylist = playlistData["playlist"];
+        if (objectKey != null) {
+          unawaited(
+            cacheManager.putFile(
+              _getCacheKey(objectKey),
+              Uint8List.fromList(
+                (playlistData["playlist"] as String).codeUnits,
+              ),
+            ),
+          );
+        }
+      }
+
+      final videoFile = objectKey == null
+          ? null
+          : (await videoCacheManager
+                  .getFileFromCache(_getVideoPreviewKey(objectKey)))
+              ?.file;
+      if (videoFile == null) {
+        final response2 = await _dio.get(
+          "/files/data/preview",
+          queryParameters: {
+            "fileID": file.uploadedFileID,
+            "type": "vid_preview",
+          },
+        );
+        final previewURL = response2.data["url"];
+        if (objectKey != null) {
+          unawaited(
+            downloadAndCacheVideo(
+              previewURL,
+              _getVideoPreviewKey(objectKey),
+            ),
+          );
+        }
+        finalPlaylist =
+            finalPlaylist.replaceAll('\noutput.ts', '\n$previewURL');
+      } else {
+        finalPlaylist =
+            finalPlaylist.replaceAll('\noutput.ts', '\n${videoFile.path}');
+      }
+
+      final tempDir = await getTemporaryDirectory();
+      final playlistFile = File("${tempDir.path}/${file.uploadedFileID}.m3u8");
+      await playlistFile.writeAsString(finalPlaylist);
+      _logger.info("Writing playlist to ${playlistFile.path}");
+      return playlistFile;
+    } catch (_) {
+      rethrow;
+    }
+  }
+
+  Future downloadAndCacheVideo(String url, String key) async {
+    final file = await videoCacheManager.downloadFile(url, key: key);
+    return file;
+  }
+
+  Future<String> getPreviewUrl(EnteFile file) async {
+    try {
+      final response = await _dio.get(
+        "/files/data/preview",
+        queryParameters: {
+          "fileID": file.uploadedFileID,
+          "type":
+              file.fileType == FileType.video ? "vid_preview" : "img_preview",
+        },
+      );
+      return response.data["url"];
+    } catch (e) {
+      _logger.warning("Failed to get preview url", e);
+      rethrow;
+    }
+  }
+
+  // get all files after cutoff date and add it to queue for preview creation
+  // only run when video streaming is enabled
+  Future<void> putFilesForPreviewCreation() async {
+    if (!isVideoStreamingEnabled) return;
+
+    final cutoff = videoStreamingCutoff;
+    if (cutoff == null) return;
+
+    final files = await FilesDB.instance.getAllFilesAfterDate(
+      fileType: FileType.video,
+      beginDate: cutoff,
+    );
+    await Future.delayed(const Duration(seconds: 5));
```

Comment:
Please add comment/reason for these wait times.

Reply:
Removed the wait times. Reasoning was to wait for previews to be loaded, which are loaded by the ML services.